OTAnalytics/plugin_datastore/track_geometry_store/polars_geometry_store.py

```diff
@@ -0,0 +1,1396 @@
+        """
+        if not isinstance(other, PolarsTrackGeometryDataset):
+            return False
+
+        # If both are empty, they are equal
+        if self.empty and other.empty:
+            return True
+
+        # If only one is empty, they are not equal
+        if self.empty != other.empty:
+            return False
+
+        # Compare the DataFrames
+        # First, ensure both DataFrames have the same columns
+        if set(self._segments_df.columns) != set(other._segments_df.columns):
+            return False
+
+        # Sort both DataFrames to ensure consistent comparison
+        self_df = self._segments_df.sort([TRACK_ID, START_OCCURRENCE])
+        other_df = other._segments_df.sort([TRACK_ID, START_OCCURRENCE])
+
+        # Compare the DataFrames using polars frame_equal
+        try:
+            return self_df.equals(other_df)
+        except Exception:
+            # Fallback to row-by-row comparison if frame_equal fails
+            if len(self_df) != len(other_df):
+                return False
+
+            # Convert to list of dicts for comparison
+            self_rows = self_df.to_dicts()
+            other_rows = other_df.to_dicts()
+
+            return self_rows == other_rows
+
+    def track_ids_after_cut(self, section: Section) -> pl.DataFrame:
+        if not section:
+            return pl.DataFrame()
+
+        # Create a set to store the track IDs that intersect with any section
+        result = self._segments_df.select([ROW_ID, track.TRACK_ID]).with_columns(
+            pl.lit(False).alias(INTERSECTS)
+        )
+
+        # Get the coordinates of the section
+        coordinates = section.get_coordinates()
+        offset = get_section_offset(section)
+
+        if section.get_type() in [SectionType.CUTTING, SectionType.LINE]:
+            # For line sections, check if any track segment intersects with any leg
+            # of the line. A leg is formed by each consecutive pair of coordinates.
+            for i in range(len(coordinates) - 1):
+                start_x, start_y = coordinates[i].x, coordinates[i].y
+                end_x, end_y = coordinates[i + 1].x, coordinates[i + 1].y
+
+                # Find intersections with this leg of the line
+                intersections = find_line_intersections(
+                    self._segments_df,
+                    section.id.serialize(),
+                    start_x,
+                    start_y,
+                    end_x,
+                    end_y,
+                    offset,
+                )
+
+                # Add track IDs that intersect with the line to the result set
+                result = (
+                    result.join(
+                        intersections.select([ROW_ID, TRACK_ID, INTERSECTS]), on=ROW_ID
+                    )
+                    .with_columns(
+                        pl.col(INTERSECTS).or_(
+                            pl.col(f"{INTERSECTS}_right")
```

Comment:
What is the difference between INTERSECTS and INTERSECTS_right?

Reply:
`INTERSECTIONS_right` will be created by the `join` before. The already existing `INTERSECTS` will be combined with the "intersects of this section segment".